presto-main/src/main/java/com/facebook/presto/server/GracefulShutdownHandler.java

```diff
@@ -86,33 +86,14 @@ public synchronized void requestShutdown()
 
         //wait for a grace period to start the shutdown sequence
         shutdownHandler.schedule(() -> {
-            List<TaskInfo> activeTasks = getActiveTasks();
-            while (activeTasks.size() > 0) {
-                CountDownLatch countDownLatch = new CountDownLatch(activeTasks.size());
-
-                for (TaskInfo taskInfo : activeTasks) {
-                    sqlTaskManager.addStateChangeListener(taskInfo.getTaskId(), newState -> {
-                        if (newState.isDone()) {
-                            countDownLatch.countDown();
-                        }
-                    });
-                }
-
-                log.info("Waiting for all tasks to finish");
-
-                try {
-                    countDownLatch.await();
-                }
-                catch (InterruptedException e) {
-                    log.warn("Interrupted while waiting for all tasks to finish");
-                    currentThread().interrupt();
-                }
-
-                activeTasks = getActiveTasks();
+            waitForTasksToComplete();
```

Comment:
I think this can be done in just the worker path.

Reply:
I included this for coordinator to cover the cases where "node-scheduler.include-coordinator" is used as true. Please correct me if I am wrong here.

Reply:
I think it make sense to skip this for coordinator given query monitoring will consider tasks as well. So it becomes redundant for coordinator. So let's skip this for coordinator.